zebra-rpc/src/server/http_request_compatibility.rs

```diff
@@ -103,11 +103,13 @@ impl FixHttpRequestMiddleware {
             .replace(", \"jsonrpc\": \"1.0\"", "")
     }
 
-    /// If the `content-type` HTTP header is not present,
-    /// add an `application/json` content type header.
-    pub fn add_missing_content_type_header(headers: &mut hyper::header::HeaderMap) {
-        headers
-            .entry(hyper::header::CONTENT_TYPE)
-            .or_insert(hyper::header::HeaderValue::from_static("application/json"));
+    /// Ignore client supplied `content-type` HTTP header if any and always send requests with `application/json`.
+    ///
+    /// `application/json` is the only `content-type` accepted by the Zebra rpc endpoint.
```

Comment:
This is correct, I confirmed it by searching the `jsonrpc_http_server` GitHub for "application/json", and finding the code that validates headers.

```suggestion
    /// `application/json` is the only `content-type` accepted by the Zebra rpc endpoint:
    /// <https://github.com/paritytech/jsonrpc/blob/38af3c9439aa75481805edf6c05c6622a5ab1e70/http/src/handler.rs#L582-L584>
```

Reply:
Was added as part of https://github.com/ZcashFoundation/zebra/pull/6885/commits/6ef6969397f44c560c7c772461f7ccbc0fb6e1c9